extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/common/KubernetesPeonClient.java

```diff
@@ -42,15 +46,22 @@ public class KubernetesPeonClient
   private final KubernetesClientApi clientApi;
   private final String namespace;
   private final boolean debugJobs;
+  private final ServiceEmitter emitter;
 
-  public KubernetesPeonClient(KubernetesClientApi clientApi, String namespace, boolean debugJobs)
+  public KubernetesPeonClient(
+      KubernetesClientApi clientApi,
+      String namespace,
+      boolean debugJobs,
+      ServiceEmitter emitter
+  )
   {
     this.clientApi = clientApi;
     this.namespace = namespace;
     this.debugJobs = debugJobs;
+    this.emitter = emitter;
   }
 
-  public Pod launchPeonJobAndWaitForStart(Job job, long howLong, TimeUnit timeUnit)
+  public Pod launchPeonJobAndWaitForStart(Job job, Task task, long howLong, TimeUnit timeUnit)
```

Comment:
```suggestion
  public Pod launchPeonJobAndWaitForStart(Task task, long howLong, TimeUnit timeUnit)
```

Reply:
Agree this is better but I found change like this will make the integration test `DruidPeonClientIntegrationTest` hard to write as we push the job creation to the underlying class, there is no easy way to construct a job to suit our testing purpose. So I decide to keep the interface as it is.